src/Transform/ONNX/ConstProp.td

```diff
@@ -208,6 +220,19 @@ def AddConstProp : Pat<
     // Additional constraints (dense)
     [(IsFromDenseONNXConstantOp:$lhs), (IsFromDenseONNXConstantOp:$rhs)]>;
 
+// TODO: Expand $a to $result's shape instead of requiring ValuesHaveSameType.
```

Comment:
`$a` isn't named in the tablegen rule. You probably mean `$x `

Reply:
good catch, fixed